geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -1,33 +1,68 @@
-import { Injectable, Logger } from '@nestjs/common';
+import { BadRequestException, Injectable, Logger } from '@nestjs/common';
+import { FeatureCollection } from 'geojson';
 import { FileService } from '../files/files.service';
+
 const mapshaper = require('mapshaper');
 
+const MIN_REQUIRED_FILES = ['prj', 'dbf', 'shx', 'shp'];
 @Injectable()
 export class ShapefileService {
   private readonly logger: Logger = new Logger(ShapefileService.name);
   constructor(private readonly fileService: FileService) {}
 
   private async shapeFileToGeoJson(fileInfo: Express.Multer.File) {
+    if (
+      !(await this.fileService.areFilesInFolder(
+        fileInfo.path.replace('.zip', ''),
+        MIN_REQUIRED_FILES,
+      ))
+    ) {
+      throw new Error('Some required file mising');
+    }
+    console.log('works');
     const outputKey = `shapefile-${new Date().getTime()}.geojson`;
 
     const _geoJson = await mapshaper.applyCommands(
-      `-i ${fileInfo.path.replace('.zip', '')}/*.shp -info -o ${outputKey}`,
+      `-i no-topology snap ${fileInfo.path.replace(
+        '.zip',
+        '',
+      )}/*.shp -proj EPSG:4326 -clean rewind -info -o ${outputKey}`,
```

Comment:
how would map-shaper handle paths with spaces in them? Maybe we should wrap them with `"`

Reply:
Nice catch! I've actually had some issues there, I decided to delete whitespaces in MulterOptions
Changes here 61dd5b268ff1f26718f885c3b84a22e9d8354bd0